counter/example.go

```diff
@@ -0,0 +1,59 @@
+package counter
```

Comment:
After trying this I remembered why it doesn't work. The example_test.go gets picked up by go test in the travis and fails since you have to explicitly set an environment variable to choose an implementation to test with.